src/features/user.ts

```diff
@@ -0,0 +1,274 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { configureStore } from "@reduxjs/toolkit";
+import firebase, { provider, ref } from "../FirebaseContext";
+import { ItemType } from "../types/ListItem";
+import { TabType } from "../types/Tab";
+import { TagType } from "../types/Tag";
+import { UserDataType, UserType } from "../types/User";
+import { loadItems } from "./items";
+import { reload } from "./loading";
+import { loadTabs, setUser } from "./tabs";
+import { loadKnownTags } from "./tags";
+
+export const LOGIN_REQUEST = "LOGIN_REQUEST";
+export type LoginRequestAction = {
+  type: typeof LOGIN_REQUEST;
+};
+export const loginRequest: () => LoginRequestAction = () => {
+  return {
+    type: LOGIN_REQUEST,
+  };
+};
+
+export const LOGIN_SUCCESS = "LOGIN_SUCCESS";
+export type LoginSuccessAction = {
+  type: typeof LOGIN_SUCCESS;
+  user: UserType;
+};
+export const loginSuccess: (user: UserType) => LoginSuccessAction = (user) => {
+  return {
+    type: LOGIN_SUCCESS,
+    user,
+  };
+};
+
+export const LOGIN_FAILURE = "LOGIN_FAILURE";
+export type LoginFailureAction = {
+  type: typeof LOGIN_FAILURE;
+};
+export const loginFailure: () => LoginFailureAction = () => {
+  return {
+    type: LOGIN_FAILURE,
+  };
+};
+
+export const LOGOUT_REQUEST = "LOGOUT_REQUEST";
+export type LogoutRequestAction = {
+  type: typeof LOGOUT_REQUEST;
+};
+export const logoutRequest: () => LogoutRequestAction = () => {
+  return {
+    type: LOGOUT_REQUEST,
+  };
+};
+
+export const LOGOUT_SUCCESS = "LOGOUT_SUCCESS";
+export type LogoutSuccessAction = {
+  type: typeof LOGOUT_SUCCESS;
+};
+export const logoutSuccess: () => LogoutSuccessAction = () => {
+  return {
+    type: LOGOUT_SUCCESS,
+  };
+};
+
+export const LOGOUT_FAILURE = "LOGOUT_FAILURE";
+export type LogoutFailureAction = {
+  type: typeof LOGOUT_FAILURE;
+};
+export const logoutFailure: () => LogoutFailureAction = () => {
+  return {
+    type: LOGOUT_FAILURE,
+  };
+};
+
+export const VERIFY_REQUEST = "VERIFY_REQUEST";
+export type VerifyRequestAction = {
+  type: typeof VERIFY_REQUEST;
+};
+export const verifyRequest: () => VerifyRequestAction = () => {
+  return {
+    type: VERIFY_REQUEST,
+  };
+};
+
+export const VERIFY_SUCCESS = "VERIFY_SUCCESS";
+export type VerifySuccessAction = {
+  type: typeof VERIFY_SUCCESS;
+};
+export const verifySuccess: () => VerifySuccessAction = () => {
+  return {
+    type: VERIFY_SUCCESS,
+  };
+};
+
+export type UserState = UserDataType;
+const initialUserState: UserState = {
```

Comment:
Do we really need all these properties? E.g. how about transforming all those error ones into one `error: string`?